exo-go/src/app_runner/index.go

```diff
@@ -82,23 +89,49 @@ func (a *AppRunner) Shutdown(shutdownConfig types.ShutdownConfig) error {
 
 // Start runs the application and returns the process and returns an error if any
 func (a *AppRunner) Start() error {
-	process, err := dockerCompose.RunAllImages(a.getEnv(), a.DockerConfigLocation, a.write)
+	dependencyNames, err := appConfigHelpers.GetAllDependencyNames(a.AppDir, a.AppConfig)
 	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("Failed to run images\nOutput: %s\nError: %s\n", process.Output, err))
+		return err
 	}
-	onlineTexts, err := a.compileOnlineTexts()
+	serviceNames := appConfigHelpers.GetServiceNames(a.AppConfig.Services)
+	serviceConfigs, err := serviceConfigHelpers.GetServiceConfigs(a.AppDir, a.AppConfig)
 	if err != nil {
 		return err
 	}
-	for role, onlineText := range onlineTexts {
-		if err = a.waitForOnlineText(process, role, onlineText); err != nil {
+	if err := a.startDependencies(dependencyNames, serviceConfigs); err != nil {
+		return err
+	}
+	return a.startServices(serviceNames, serviceConfigs)
+}
+
+func (a *AppRunner) startDependencies(dependencyNames []string, serviceConfigs map[string]types.ServiceConfig) error {
+	dependencyOnlineTexts := a.compileDependencyOnlineTexts(serviceConfigs)
+	process, err := dockerCompose.RunImages(dependencyNames, a.getEnv(), a.DockerConfigLocation, a.write)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to run dependencies\nOutput: %s\nError: %s\n", process.Output, err))
+	}
+	for dependencyName, onlineText := range dependencyOnlineTexts {
+		if err := a.waitForOnlineText(process, dependencyName, onlineText); err != nil {
 			return err
 		}
 	}
-	if err == nil {
-		a.write("all services online")
+	a.write("all dependencies online")
+	return nil
+}
+
+func (a *AppRunner) startServices(serviceNames []string, serviceConfigs map[string]types.ServiceConfig) error {
+	serviceOnlineTexts := a.compileServiceOnlineTexts(serviceConfigs)
+	process, err := dockerCompose.RunImages(serviceNames, a.getEnv(), a.DockerConfigLocation, a.write)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to run services\nOutput: %s\nError: %s\n", process.Output, err))
+	}
+	for role, onlineText := range serviceOnlineTexts {
+		if err := a.waitForOnlineText(process, role, onlineText); err != nil {
+			return err
+		}
 	}
-	return err
+	a.write("all services online")
+	return nil
```

Comment:
`startDependencies` and `startServices`are very repetitive. I think the two could be dried up with something like:
```go
func startImages(imageNames []string, imageOnlineTexts map[string]string, identifier string) error {
  // identifier used when wrapping errors and for the final log message 
}
```